### PR TITLE
Clean up IOUAmount page

### PR DESCRIPTION
**Pull request by thienlnam**
### Details

1. Removes the ActivityIndicator
2. Autofocuses on input on web
3. Remember the amount entered when we return to the page from later steps

### Fixed Issues

Fixes https://github.com/Expensify/Expensify/issues/163041

### Tests / QA

1. Go to someone's chat and click the + button and hit 'Request Money'
2. (On Web/Desktop), make sure the amount field is focused on (entering any text automatically adds it and the cursor is blinking)
Screen Shot 2021-05-06 at 2 08 43 PM

3. (On Web/Desktop) Enter any amount and hit the 'Enter' button on your keyboard.  Make sure that it takes you to the next screen.
3. (Other platforms) Enter in any amount and hit next
4. Go back and ensure that you still see the amount you entered before
Screen Shot 2021-05-06 at 1 58 32 PM

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

Screen Shot 2021-05-06 at 2 34 45 PM

#### Mobile Web

Screen Shot 2021-05-06 at 2 53 04 PM

#### Desktop

Screen Shot 2021-05-06 at 2 35 56 PM

#### iOS

Screen Shot 2021-05-06 at 2 40 10 PM

#### Android

Screen Shot 2021-05-06 at 2 48 43 PM

### CODE REVIEW

File: src/pages/iou/steps/IOUAmountPage.js
Comment (thienlnam) [inline src/pages/iou/steps/IOUAmountPage.js:58]: We were already trying to focus the input here but it wasn't working - the only workaround that worked was adding a setTimeout for anything > 300.  I also tried the autoFocus prop on the textInput but that didn't work either

**Comment by thienlnam** [comment]
Got a request in the original issue to add a feature: https://github.com//Expensify/issues/163041#issuecomment-834285328 

Putting on hold while I take care of that

**Comment by thienlnam** [comment]
Alright, ready to be reviewed! @stitesExpensify

File: src/pages/iou/steps/IOUAmountPage.js
Comment (Jag96) [inline src/pages/iou/steps/IOUAmountPage.js:58]: Is there no other way to do this without a setTimeout?

**Review by Jag96 — COMMENTED** [review]
I think we should avoid the setTimeout call if possible

File: src/pages/iou/steps/IOUAmountPage.js
Comment (thienlnam) [reply src/pages/iou/steps/IOUAmountPage.js]: Updated! Looks like the iou not being done loading was the issue - the componentDidMount ran before the iou was actually finished loading so I had to use a componentDidMount to watch for the prop change and then focus on the input

**Review by Jag96 — COMMENTED** [review]
Looking good, just one comment and one question!

File: src/pages/iou/steps/IOUAmountPage.js
Comment (jasperhuangg) [reply src/pages/iou/steps/IOUAmountPage.js:51]: +1

File: src/pages/iou/steps/IOUAmountPage.js
Comment (jasperhuangg) [reply src/pages/iou/steps/IOUAmountPage.js:73]: I also agree. Unless there are other event handlers that may be added in the future (and even then this is an early optimization), I think this just creates an extra level of indirection considering how simple the logic is.

**Comment by thienlnam** [comment]
Thanks for the comments, updated! I was also able to find a cleaner way to handle the focus

**Review by jasperhuangg — DISMISSED** [review]
Looks good!

**Comment by thienlnam** [comment]
Fixed merge conflicts!

**Review by Jag96 — CHANGES_REQUESTED** [review]
Getting a crash on iOS because `document` doesn't exist in the mobile context:

**Comment by stitesExpensify** [comment]
> Getting a crash on iOS because document doesn't exist in the mobile context:

Should we split this file into `index.js` and `index.native.js`?

**Comment by thienlnam** [comment]
Nice catch, @stitesExpensify yup about to do that - currently testing things that don't need to be included in the native version

**Comment by thienlnam** [comment]
Split into index and index.native.js except after the latest merge the focus is no longer working 😢  Asked about it in slack https://expensify.slack.com/archives/C01GTK53T8Q/p1620935953167600

File: src/pages/iou/steps/IOUAmountPage/index.js
Comment (tgolen) [inline src/pages/iou/steps/IOUAmountPage/index.js:138]: Indentation is off here

File: src/pages/iou/steps/IOUAmountPage/index.js
Comment (tgolen) [inline src/pages/iou/steps/IOUAmountPage/index.js:137]: Why isn't the `autoFocus` property being used? I think that's specifically what it's for.

File: src/pages/iou/steps/IOUAmountPage/index.js
Comment (tgolen) [inline src/pages/iou/steps/IOUAmountPage/index.js:154]: Indentation is off here too.

**Review by tgolen — CHANGES_REQUESTED** [review]

File: src/pages/iou/steps/IOUAmountPage/index.js
Comment (thienlnam) [reply src/pages/iou/steps/IOUAmountPage/index.js]: Ran into that weird eslint bug but did the workaround

File: src/pages/iou/steps/IOUAmountPage/index.js
Comment (thienlnam) [reply src/pages/iou/steps/IOUAmountPage/index.js]: Ran into that weird eslint bug but did the workaround

**Comment by thienlnam** [comment]
Updated, ready to go now! There was a weird eslint bug and fixing the neighbors fixed it 

cc @stitesExpensify @Jag96 @jasperhuangg @tgolen

File: ios/Podfile.lock
Comment (stitesExpensify) [inline ios/Podfile.lock:709]: Was this intentional?

Comment (thienlnam) [reply ios/Podfile.lock]: Nope 🤦

**Comment by thienlnam** [comment]
Removed the accidental podfile change 👍

File: src/pages/iou/steps/IOUAmountPage/index.native.js
Comment (tgolen) [inline src/pages/iou/steps/IOUAmountPage/index.native.js:68]: Is this comment out of date now?

**Review by tgolen — CHANGES_REQUESTED** [review]
Hm, I don't know where I said this before, I swear I said it, but can't find it now :(

I'm not a fan of these files being split into index/index.native like they are currently. This will be the first time that an entire page is split between platforms, and it really creates a lot of repeated code. I think a solution should be found that only abstracts the absolute least amount of logic in the platform-specific files as possible. This will decrease the cost to maintain these files in the future as it's not very clear what logic is different between the two files.

**Comment by thienlnam** [comment]
I agree with you on all those points, the only logic difference in the files was the keydown event handler for Web/Desktop, which was having `document is undefined` errors on mobile.  I thought the way to handle that would be to split them into different files but it actually turns out there is a KeyboardShortcut component that handles precisely this problem so thanks for bringing that up 👍

**Review by tgolen — DISMISSED** [review]
Ah, that looks great! Thank you

**Review by Jag96 — CHANGES_REQUESTED** [review]
I'm seeing another crash on iOS after step 4 `Enter in any amount and hit next`. Do you see the same @thienlnam?

```
TypeError: undefined is not an object (evaluating 'option.participantsList.length')
This error is located at:
    in OptionRow
    in OptionRow (at OptionsList.js:160)
```

**Comment by thienlnam** [comment]
@Jag96 I'm seeing those same errors but they're happening on prod and unrelated to these changes https://github.com/Expensify/Expensify.cash/issues/2969.  Just updated back to those block comments

**Review by Jag96 — APPROVED** [review]

**Comment by thienlnam** [comment]
Going to self merge since we got 2 approvals

**Pull request merged**
